plugins/playbackProfile.ts

```diff
@@ -0,0 +1,264 @@
+import { Context } from '@nuxt/types/app';
+import { browserDetector } from './browserDetection';
+import {
+  getSupportedMP4VideoCodecs,
+  hasVp8Support
+} from '~/utils/mp4VideoFormats';
+import { getSupportedMP4AudioCodecs } from '~/utils/mp4AudioFormats';
+import { getSupportedVPXVideoCodecs } from '~/utils/vpxVideoFormats';
+import { getSupportedWebMAudioCodecs } from '~/utils/webmAudioFormats';
+import { getSupportedAudioCodecs } from '~/utils/audioFormats';
+import { canPlayNativeHls, hasMkvSupport } from '~/utils/transcodingFormats';
+import { getHlsVideoCodecs, getHlsAudioCodecs } from '~/utils/hlsFormats';
+import { getCodecProfiles } from '~/utils/codecProfiles';
+import {
+  DeviceProfile,
+  DirectPlayProfile,
+  DlnaProfileType,
+  TranscodingProfile,
+  EncodingContext,
+  SubtitleProfile,
+  SubtitleDeliveryMethod,
+  ResponseProfile
+} from '~/api';
+import { PluginInjection } from '~/types/utils';
+
+const physicalAudioChannels = browserDetector.isTv() ? 6 : 2;
+
+const videoTestElement = document.createElement('video');
+
+/**
+ *
+ */
+function getDirectPlayProfiles(): Array<DirectPlayProfile> {
+  const DirectPlayProfiles = [];
+
+  const mp4VideoCodecs = getSupportedMP4VideoCodecs(videoTestElement);
+  const mp4AudioCodecs = getSupportedMP4AudioCodecs(videoTestElement);
+  const vpxVideoCodecs = getSupportedVPXVideoCodecs(videoTestElement);
+  const webmAudioCodecs = getSupportedWebMAudioCodecs();
+
+  DirectPlayProfiles.push({
+    Container: 'webm',
+    Type: DlnaProfileType.Video,
+    AudioCodec: webmAudioCodecs.join(','),
+    VideoCodec: vpxVideoCodecs.join(',')
+  });
+
+  DirectPlayProfiles.push({
+    Container: 'mp4,m4v',
+    Type: DlnaProfileType.Video,
+    VideoCodec: mp4VideoCodecs.join(','),
+    AudioCodec: mp4AudioCodecs.join(',')
+  });
+
+  const supportedAudio = [
+    'opus',
+    'mp3',
+    'mp2',
+    'aac',
+    'flac',
+    'alac',
+    'webma',
+    'wma',
+    'wav',
+    'ogg',
+    'oga'
+  ];
```

Comment:
Do these need to be tested? Or can it be assumed that all of the formats are supported

Reply:
I think it's assumed that they all work. This section in particular is identical to jf-web.

Reply:
Looking at caniuse.com, most of the above are supported, but: 
- ogg isn't suppported on safari
- oga has very patchy support

This can probably be left for now, and proper checking added at a later date?